### ISSUE TITLE
Performance difference between Spark and Sparklyr

### ISSUE THREAD

**Issue by lalitnookala** [rstudio/sparklyr#312]
Good Afternoon all -

This is NOT an issue but a question to understand how Sparklyr works.

SCENARIO:
Take the MNIST data from https://www.kaggle.com/c/digit-recognizer/data. Run PCA on the training data to reduce the dimension of the dataset to 150 features. Using PrComp from R is fine for it. Reformulate the data to have the label and the 150 PCs (Principal Components) as a new data file. In order to test and validate split the data in training set and test set (29K and 13K respectively) 

TEST:

On the above formed new dataset, run neural network with 150 input nodes, 5 nodes in the hidden layer, 10 output nodes using Spark REPL and Sparklyr. Additional features to be set: seed = 1234L, max iterations = 100

RESULTS:

Spark's REPL (Scala) takes 427 seconds to execute.
Sparklyr takes only 141 seconds to execute.

The results are same 67.2% accuracy. 

QUESTION:

How is it possible for Sparklyr to finish the job in that short amount of time when compared to Scala code? The code that I have written on both Spark and Sparklyr is similar:
1. Initialize the Neural Network with the appropriate options such as seed, layers and max iteration
2. Fit the training data
3. Transform the test data.

Other computations such as evaluation of the classifier and computation of accuracy and error in not part of the benchmark time.

Please let me know your thoughts on this matter.

Sincerely
Lalit Nookala

**Comment by javierluraschi** [comment]
@lalitnookala `sparklyr` eagerly caches tables into memory to give a performance boost by default, if you are using `copy_to` or `spark_read_csv`, etc. Try setting the `memory` parameter to `FALSE`. That might slowdown `sparklyr` enough to match the Scala REPL.

**Comment by lalitnookala** [comment]
Javier,

Ah! That makes sense and excellent design choice. I wonder if this is the effect of Project Tungsten's Cache-aware computation.

I will try setting memory parameter to FALSE in Sparklyr and also separately try caching table on the Spark side to see if one catches up with the other. 

Thank you very much for your reply.

**Comment by javierluraschi** [comment]
Sounds good @lalitnookala, thanks for closing this issue.
